pkg/workqueue/gcs/README.md

```diff
@@ -0,0 +1,52 @@
+# GCS Workqueue Implementation
+
+This package implements a Google Cloud Storage (GCS) backed workqueue that provides reliable, persistent task processing with state management.
+
+## Bucket Organization
+
+The GCS workqueue uses object prefixes to organize tasks by their state within a single bucket:
+
+### Prefixes
+
+- **`queued/`** - Tasks waiting to be processed
+- **`in-progress/`** - Tasks currently being processed by a worker
+- **`dead-letter/`** - Tasks that have failed after exceeding maximum retry attempts
+
+### State Transitions
+
+```
+queued/{key} → in-progress/{key} → [completed] (deleted)
+                      ↓
+                 dead-letter/{key} (on failure)
+                      ↑
```

Comment:
The direction of this seems incorrect? 🤔

Reply:
Is it? dead-letter should always be the sync here, no?

Formatting is a little weird, but I think it's correct?